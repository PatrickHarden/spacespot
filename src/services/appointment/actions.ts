import {
  AppointmentAction,
  Appointment,
  EditAppointment,
  NewAppointment,
} from './types'
import constants from './constants'

const get = (): AppointmentAction => ({
  type: constants.GET,
})
const getSuccess = (payload: Array<Appointment>) => ({
  type: constants.GET_SUCCESS,
  payload,
})
const getError = (payload: string): AppointmentAction => ({
  type: constants.GET_ERROR,
  payload,
})
const update = (): AppointmentAction => ({
  type: constants.UPDATE,
})

const create = (payload: NewAppointment): AppointmentAction => ({
  type: constants.CREATE,
  payload,
})

const createSuccess = (payload: Appointment) => ({
  type: constants.CREATE_SUCCESS,
  payload,
})
const createError = (payload: string): AppointmentAction => ({
  type: constants.CREATE_ERROR,
  payload,
})

const accept = (payload: string): AppointmentAction => ({
  type: constants.ACCEPT,
  payload,
})

const acceptSuccess = (payload: Appointment) => ({
  type: constants.ACCEPT_SUCCESS,
  payload,
})

const acceptError = (payload: string): AppointmentAction => ({
  type: constants.ACCEPT_ERROR,
  payload,
})

const cancel = (payload: string): AppointmentAction => ({
  type: constants.CANCEL,
  payload,
})

const cancelSuccess = (payload: Appointment) => ({
  type: constants.CANCEL_SUCCESS,
  payload,
})

const cancelError = (payload: string): AppointmentAction => ({
  type: constants.CANCEL_ERROR,
  payload,
})

const change = (payload: EditAppointment): AppointmentAction => ({
  type: constants.CHANGE,
  payload,
})

const changeSuccess = (payload: Appointment) => ({
  type: constants.CHANGE_SUCCESS,
  payload,
})

const changeError = (payload: string): AppointmentAction => ({
  type: constants.CHANGE_ERROR,
  payload,
})
export default {
  get,
  getSuccess,
  getError,
  update,
  create,
  createSuccess,
  createError,
  accept,
  acceptSuccess,
  acceptError,
  cancel,
  cancelSuccess,
  cancelError,
  change,
  changeSuccess,
  changeError,
}
