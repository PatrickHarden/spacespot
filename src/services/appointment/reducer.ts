import constants from './constants'
import { AppointmentAction, AppointmentState, Appointment } from './types'

const initialState: AppointmentState = {}

const reducer = (
  state: AppointmentState = initialState,
  action: AppointmentAction,
): AppointmentState => {
  switch (action.type) {
    case constants.GET_ERROR:
      return { ...state, error: action.payload as string }
    case constants.GET_SUCCESS:
      return { ...state, data: action.payload as Array<Appointment> }
    case constants.CREATE_SUCCESS:
      return {
        ...state,
        data: [...(state.data || []), action.payload as Appointment],
      }
    default:
      return state
  }
}
export default reducer
