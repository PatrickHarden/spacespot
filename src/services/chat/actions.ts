import { ChatAction, ChatResponse, ChatMessage } from './types'
import constants from './constants'

const get = (): ChatAction => ({
  type: constants.GET,
})
const getSuccess = (payload: ChatResponse) => ({
  type: constants.GET_SUCCESS,
  payload,
})
const getError = (payload: string): ChatAction => ({
  type: constants.GET_ERROR,
  payload,
})

const newMessage = (payload: string) => ({
  type: constants.NEW_MESSAGE,
  payload,
})
const newMessageSuccess = (payload: ChatMessage) => ({
  type: constants.NEW_MESSAGE_SUCCESS,
  payload,
})
const newMessageError = (payload: string) => ({
  type: constants.NEW_MESSAGE_ERROR,
  payload,
})
export default {
  get,
  getSuccess,
  getError,
  newMessage,
  newMessageSuccess,
  newMessageError,
}
