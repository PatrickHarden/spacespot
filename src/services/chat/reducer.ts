import { get } from 'lodash'
import constants from './constants'
import { ChatAction, ChatState, ChatResponse, ChatMessage } from './types'

const initialState: ChatState = {}
const addNewMessage = (
  data: ChatResponse,
  payload: ChatMessage,
): ChatResponse => {
  const messagesOld: Array<ChatMessage> = get(data, 'messages')
  return { ...data, messages: [...messagesOld, payload] }
}
const reducer = (
  state: ChatState = initialState,
  action: ChatAction,
): ChatState => {
  switch (action.type) {
    case constants.GET:
      return { ...state }
    case constants.GET_ERROR:
      return { ...state, error: action.payload as string }
    case constants.GET_SUCCESS:
      return { ...state, data: action.payload as ChatResponse }
    case constants.NEW_MESSAGE_SUCCESS:
      return {
        ...state,
        data: addNewMessage(
          state.data as ChatResponse,
          action.payload as ChatMessage,
        ),
      }
    default:
      return state
  }
}
export default reducer
