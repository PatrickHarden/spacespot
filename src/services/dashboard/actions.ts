import constants from './constants'
import { Space } from 'services/space/types'

const dashboardInit = () => ({
  type: constants.DASHBOARD_INIT,
})

const dashboardSuccess = (payload: {
  buildings: Array<Space>
  spaces: { [key: string]: Array<Space> }
}) => ({
  type: constants.DASHBOARD_SUCCESS,
  payload,
})

const dashboardError = (payload: string) => ({
  type: constants.DASHBOARD_ERROR,
  payload,
})

const pendingSpaceDelete = (payload: string) => ({
  type: constants.DASHBOARD_PENDING_SPACE_DELETE,
  payload,
})

export default {
  dashboardInit,
  dashboardSuccess,
  dashboardError,
  pendingSpaceDelete,
}
