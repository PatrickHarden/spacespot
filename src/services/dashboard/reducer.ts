import constants from './constants'
import { DashboardState } from './types'

const initialState: DashboardState = {
  initOK: false,
  buildings: [],
  spaces: {},
  isLoading: false,
  pendingDelete: [],
}

const reducer = (
  state: DashboardState = initialState,
  action: any,
): DashboardState => {
  switch (action.type) {
    case constants.DASHBOARD_SUCCESS:
      return {
        ...state,
        buildings: action.payload.buildings,
        spaces: action.payload.spaces,
        isLoading: false,
        initOK: true,
      }
    case constants.DASHBOARD_ERROR:
      return {
        ...state,
        isLoading: false,
        error: action.payload as string,
      }
    case constants.DASHBOARD_INIT:
      return {
        ...state,
        buildings: [],
        spaces: {},
        isLoading: true,
        pendingDelete: state.pendingDelete,
      }
    case constants.DASHBOARD_PENDING_SPACE_DELETE:
      return {
        ...state,
        pendingDelete: [...state.pendingDelete, action.payload as string],
      }
    default:
      return state
  }
}

export default reducer
