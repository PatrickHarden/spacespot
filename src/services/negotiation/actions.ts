import types from './constants'
import {
  NegotiationAPIResponse,
  NegotiationTerm,
  FitoutOptionProps,
  SelectedFitoutOption,
  NegotiationStatus,
  NegotiationStatusResponse,
  SignerInfoDTO,
  SignerDetailDTO,
} from './types'

const get = () => ({
  type: types.GET,
})

const getSuccess = (payload: NegotiationAPIResponse) => ({
  type: types.GET_SUCCESS,
  payload,
})

const getError = (payload: string) => ({
  type: types.GET_ERROR,
  payload,
})

const getFields = () => ({
  type: types.GET_FIELDS,
})

const getFieldsSuccess = (payload: any) => ({
  type: types.GET_FIELDS_SUCCESS,
  payload,
})

const getFieldsError = (payload: string) => ({
  type: types.GET_FIELDS_ERROR,
  payload,
})

const putCounter = (payload: Array<NegotiationTerm>) => ({
  type: types.PUT_COUNTER,
  payload,
})

const putCounterSuccess = (payload: any) => ({
  type: types.PUT_COUNTER_SUCCESS,
  payload,
})

const putCounterError = (payload: string) => ({
  type: types.PUT_COUNTER_ERROR,
  payload,
})

const accept = () => ({
  type: types.ACCEPT,
})
const acceptSuccess = (payload: any): any => ({
  type: types.ACCEPT_SUCCESS,
  payload,
})
const acceptError = (payload: string): any => ({
  type: types.ACCEPT_ERROR,
  payload,
})

const reject = () => ({
  type: types.REJECT,
})
const rejectSuccess = (payload: any): any => ({
  type: types.REJECT_SUCCESS,
  payload,
})
const rejectError = (payload: string): any => ({
  type: types.REJECT_ERROR,
  payload,
})

const getCustomFitout = () => ({
  type: types.GET_CUSTOM_FITOUT,
})

const getCustomFitoutSuccess = (payload: any) => ({
  type: types.GET_CUSTOM_FITOUT_SUCCESS,
  payload,
})

const getCustomFitoutError = (payload: string) => ({
  type: types.GET_CUSTOM_FITOUT_ERROR,
  payload,
})

const putCustomFitout = (payload: FitoutOptionProps) => ({
  type: types.PUT_CUSTOM_FITOUT,
  payload,
})

const putCustomFitoutSuccess = (payload: any) => ({
  type: types.PUT_CUSTOM_FITOUT_SUCCESS,
  payload,
})

const putCustomFitoutError = (payload: string) => ({
  type: types.PUT_CUSTOM_FITOUT_ERROR,
  payload,
})

const saveSpecialProvisions = (payload: string) => ({
  type: types.SAVE_SPECIAL_PROVISIONS,
  payload,
})

const saveSpecialProvisionsSuccess = () => ({
  type: types.SAVE_SPECIAL_PROVISIONS_SUCCESS,
})

const saveSpecialProvisionsError = (payload: string) => ({
  type: types.SAVE_SPECIAL_PROVISIONS_ERROR,
  payload,
})

const acceptSpecialProvisions = () => ({
  type: types.ACCEPT_SPECIAL_PROVISIONS,
})

const rejectSpecialProvisions = () => ({
  type: types.REJECT_SPECIAL_PROVISIONS,
})

const selectFitout = (payload: SelectedFitoutOption) => ({
  type: types.SELECT_FITOUT,
  payload,
})
const selectFitoutSuccess = (payload: NegotiationAPIResponse) => ({
  type: types.SELECT_FITOUT_SUCCESS,
  payload,
})
const selectFitoutError = (payload: string) => ({
  type: types.SELECT_FITOUT_ERROR,
  payload,
})
const deleteFitout = (payload: string) => ({
  type: types.DELETE_FITOUT,
  payload,
})
const deleteFitoutSuccess = (payload: string) => ({
  type: types.DELETE_FITOUT_SUCCESS,
  payload,
})
const deleteFitoutError = (payload: string) => ({
  type: types.DELETE_FITOUT_ERROR,
  payload,
})
const putNegotiationStatus = (payload: NegotiationStatus) => ({
  type: types.PUT_SIGN_LEASE_STATUS,
  payload,
})
const putNegotiationStatusSuccess = (payload: NegotiationStatusResponse) => ({
  type: types.PUT_SIGN_LEASE_STATUS_SUCCESS,
  payload,
})
const putNegotiationStatusError = (payload: string) => ({
  type: types.PUT_SIGN_LEASE_STATUS_ERROR,
  payload,
})

const postSignerInfo = (payload: SignerInfoDTO) => ({
  type: types.POST_SIGNER_INFO,
  payload,
})
const putSignerInfo = (payload: SignerInfoDTO) => ({
  type: types.PUT_SIGNER_INFO,
  payload,
})
const getSignerInfo = () => ({
  type: types.GET_SIGNER_INFO,
})
const signerInfoSuccess = (payload: SignerInfoDTO) => ({
  type: types.SIGNER_INFO_SUCCESS,
  payload,
})
const signerInfoError = (payload: string) => ({
  type: types.SIGNER_INFO_ERROR,
  payload,
})
const acceptLease = () => ({
  type: types.ACCEPT_LEASE,
})

const getSigners = () => ({
  type: types.SIGNERS,
})
const signersSuccess = (payload: Array<SignerDetailDTO>) => ({
  type: types.SIGNERS_SUCCESS,
  payload,
})
const signersError = (payload: string) => ({
  type: types.SIGNERS_ERROR,
  payload,
})

const getPreviewLease = () => ({
  type: types.GET_PREVIEW_LEASE,
})
const getPreviewLeaseSuccess = (payload: NegotiationStatusResponse) => ({
  type: types.GET_PREVIEW_LEASE_SUCCESS,
  payload,
})
const getPreviewLeaseError = (payload: string) => ({
  type: types.GET_PREVIEW_LEASE_ERROR,
  payload,
})

const getLeaseDocument = () => ({
  type: types.GET_LEASE_DOC,
})

export default {
  get,
  getSuccess,
  getError,
  getFields,
  getFieldsSuccess,
  getFieldsError,
  putCounter,
  putCounterSuccess,
  putCounterError,
  accept,
  acceptSuccess,
  acceptError,
  reject,
  rejectSuccess,
  rejectError,
  getCustomFitout,
  getCustomFitoutSuccess,
  getCustomFitoutError,
  putCustomFitout,
  putCustomFitoutSuccess,
  putCustomFitoutError,
  saveSpecialProvisions,
  saveSpecialProvisionsSuccess,
  saveSpecialProvisionsError,
  acceptSpecialProvisions,
  rejectSpecialProvisions,
  selectFitout,
  selectFitoutSuccess,
  selectFitoutError,
  deleteFitout,
  deleteFitoutSuccess,
  deleteFitoutError,
  putNegotiationStatus,
  putNegotiationStatusSuccess,
  putNegotiationStatusError,
  postSignerInfo,
  putSignerInfo,
  getSignerInfo,
  signerInfoSuccess,
  signerInfoError,
  acceptLease,
  getSigners,
  signersSuccess,
  signersError,
  getPreviewLease,
  getPreviewLeaseSuccess,
  getPreviewLeaseError,
  getLeaseDocument,
}
