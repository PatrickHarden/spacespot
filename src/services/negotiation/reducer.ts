import { get } from 'lodash'
import constants from './constants'
import { NegotiationState, FitoutOptionProps } from './types'

const initialState: NegotiationState = { data: undefined, loading: false }
const filterCustomFitout = (fitouts: Array<FitoutOptionProps>, id: string) =>
  fitouts.filter((fitout: FitoutOptionProps) => fitout.id !== id)

const reducer = (state = initialState, action: any) => {
  switch (action.type) {
    case constants.GET:
      return { ...state, loading: true }
    case constants.GET_ERROR:
    case constants.GET_FIELDS_ERROR:
      return { error: action.payload as string, loading: false }
    case constants.GET_SUCCESS:
      return {
        ...state,
        data: action.payload,
        loading: false,
        signers: undefined,
        signerInfo: undefined,
      }
    case constants.GET_FIELDS_SUCCESS:
      return {
        ...state,
        termsFields: action.payload,
        loading: false,
      }
    case constants.SELECT_FITOUT_SUCCESS:
    case constants.PUT_SIGN_LEASE_STATUS_SUCCESS:
      return {
        ...state,
        data: { ...state.data, ...action.payload },
      }
    case constants.GET_CUSTOM_FITOUT_SUCCESS:
      return {
        ...state,
        customFitout: action.payload,
        loading: false,
      }

    case constants.PUT_CUSTOM_FITOUT_SUCCESS:
      return {
        ...state,
        customFitout: [...get(state, 'customFitout', []), action.payload],
        loading: false,
      }

    case constants.DELETE_FITOUT_SUCCESS:
      return {
        ...state,
        customFitout: filterCustomFitout(
          get(state, 'customFitout', []),
          action.payload,
        ),
      }
    case constants.SIGNER_INFO_SUCCESS:
      return {
        ...state,
        signerInfo: action.payload,
      }
    case constants.SIGNERS_SUCCESS:
      return {
        ...state,
        signers: action.payload,
      }
    case constants.SIGNERS_ERROR:
      return {
        ...state,
        signers: undefined,
        error: action.payload as string,
      }
    case constants.SIGNER_INFO_ERROR:
      return {
        ...state,
        signerInfo: undefined,
        error: action.payload as string,
      }
    default:
      return state
  }
}

export default reducer
