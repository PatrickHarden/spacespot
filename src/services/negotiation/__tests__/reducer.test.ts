import reducer from '../reducer'
import actions from '../actions'
import {
  NegotiationAPIResponse,
  FitoutOptionProps,
  NegotiationStatusResponse,
} from '../types'

describe('Negotiation reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toEqual({
      data: undefined,
      loading: false,
    })
  })
  it('init should update state', () => {
    const resp = { loading: true }
    expect(reducer(undefined, actions.get())).toEqual(resp)
  })

  it('error should update state', () => {
    const err = { error: 'Error1' }
    expect(reducer(undefined, actions.getError('Error1'))).toEqual({
      ...err,
      loading: false,
    })
  })

  it('get success should update state', () => {
    const resp = {} as NegotiationAPIResponse
    expect(reducer(undefined, actions.getSuccess(resp))).toEqual({
      data: resp,
      loading: false,
    })
  })

  it('getFields success should update state', () => {
    const resp = {}
    expect(reducer(undefined, actions.getFieldsSuccess(resp))).toEqual({
      termsFields: resp,
      loading: false,
    })
  })

  it('custom fituot options get success should update state', () => {
    const resp: Array<FitoutOptionProps> = [
      { id: '1', isCustom: true, name: 'test', description: 'test', amount: 1 },
    ]
    expect(reducer(undefined, actions.getCustomFitoutSuccess(resp))).toEqual({
      customFitout: resp,
      loading: false,
    })
  })

  it('custom fituot options put custom fitout success should update state', () => {
    const state = {
      customFitout: [
        {
          id: '1',
          isCustom: true,
          name: 'test',
          description: 'test',
          amount: 1,
        },
      ],
      loading: false,
    }
    const resp: FitoutOptionProps = {
      id: '2',
      isCustom: true,
      name: 'test2',
      description: 'test2',
      amount: 2,
    }

    expect(
      reducer(state, actions.putCustomFitoutSuccess(resp)),
    ).toMatchSnapshot()
  })
  it('custom fituot options delete custom fitout success should update state', () => {
    const state = {
      customFitout: [
        {
          id: '1',
          isCustom: true,
          name: 'test',
          description: 'test',
          amount: 1,
        },
        {
          id: '2',
          isCustom: true,
          name: 'test2',
          description: 'test2',
          amount: 2,
        },
      ],
      loading: false,
    }
    const resp = '2'

    expect(reducer(state, actions.deleteFitoutSuccess(resp))).toMatchSnapshot()
  })
  it('custom fituot options select fitout success should update state', () => {
    const resp = {} as NegotiationAPIResponse
    expect(reducer(undefined, actions.selectFitoutSuccess(resp))).toEqual({
      data: resp,
      loading: false,
    })
  })
  it('negotiation status success should update state', () => {
    const resp = {
      negotiationId: 165,
      peer1SignLeaseStatus: 'TermsAgreed',
      peer2SignLeaseStatus: 'TermsAgreed',
    } as NegotiationStatusResponse
    expect(
      reducer(undefined, actions.putNegotiationStatusSuccess(resp)),
    ).toEqual({
      data: resp,
      loading: false,
    })
  })
})
