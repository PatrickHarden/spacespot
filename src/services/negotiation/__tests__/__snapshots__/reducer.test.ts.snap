// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Negotiation reducer custom fituot options delete custom fitout success should update state 1`] = `
Object {
  "customFitout": Array [
    Object {
      "amount": 1,
      "description": "test",
      "id": "1",
      "isCustom": true,
      "name": "test",
    },
  ],
  "loading": false,
}
`;

exports[`Negotiation reducer custom fituot options put custom fitout success should update state 1`] = `
Object {
  "customFitout": Array [
    Object {
      "amount": 1,
      "description": "test",
      "id": "1",
      "isCustom": true,
      "name": "test",
    },
    Object {
      "amount": 2,
      "description": "test2",
      "id": "2",
      "isCustom": true,
      "name": "test2",
    },
  ],
  "loading": false,
}
`;
